keps/2941-DRA/README.md

```diff
@@ -0,0 +1,729 @@
+
+## Motivation
+
+Dynamic Resource Allocation (DRA) provides the groundwork for more sophisticated device allocations to Pods.
+Quota management is about enforcing rules around the use of resources.
+For example, GPUs are resource constrained and a popular request is the ability to enforce fair sharing of GPU resources.
+With these devices, many users want access and sometimes some users want the ability to preempt other users if their
+workloads have a higher priority. Kueue provides support for this.
+
+DRA provides a future where users could schedule partitionable GPU devices (MIG) or time slicing. As devices gain a 
+more robust way to schedule, it is important to walk through how support of DRA will work with Kueue.
+
+### Background
+
+DRA has four APIs that are relevant for a Kueue:
+
+- ResourceClaims
+- ResourceClaimTemplates
+- DeviceClasses
+- ResourceSlices
+
+#### DRA Example
+
+The easiest way to test DRA is to use [dra example driver repository](https://github.com/kubernetes-sigs/dra-example-driver). Cloning that repo and running 
+`make setup-e2e` will create a Kind cluster with the DRA feature gate and install a mock dra driver. This does not use
+actual GPUs so it is perfect for a test environment for exploring Kueue and DRA integration.
+
+#### Workload Example
+
+An example workload that uses DRA:
+
+```yaml
+---
+
+apiVersion: resource.k8s.io/v1alpha3
+kind: ResourceClaimTemplate
+metadata:
+  namespace: gpu-test1
+  name: single-gpu
+spec:
+  spec:
+    devices:
+      requests:
+      - name: gpu
+        deviceClassName: gpu.example.com
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: gpu-test1
+  name: job0
+  labels:
+    app: job
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: ctr0
+        image: ubuntu:22.04
+        command: ["bash", "-c"]
+        args: ["export; sleep 9999"]
+        resources:
+          claims:
+          - name: gpu
+          requests:
+            cpu: 1
+            memory: "200Mi"
+      resourceClaims:
+      - name: gpu
+        resourceClaimTemplateName: single-gpu
+```
+
+#### Example Driver Cluster Resources
+
+The dra-example-driver creates a ResourceSlice for each node and a DeviceClass named `gpu.example.com` for the entire 
+cluster.
+
+##### ResourceSlices
+
+ResourceSlices are meant for communication between drivers and the control planes. These are not expected to be used for
+workloads.
+
+##### DeviceClasses
+
+Each driver creates a device class and every resource claim will reference the device class. The dra-example-driver has
+a simple device class named `gpu.example.com`. This will be the way to enforce quota limits.
+
+### Goals
+
+- Users can submit workloads using resource claims and Kueue can monitor the usage.
+- Admins can enforce the quota for number of devices for a given DeviceClass.
+
+
+### Non-Goals
```

Comment:
Additionally, I'm suspecting the first Kueue with the DRA feature does not have compatibility for the MultiKueue.
Because the ResourceClaimTemplate and ResourceClaim responsibilities are out of Kueue.

In other words, batch users need access to all clusters and create a ResourceClaimTemplate in their namespaces before they submit Jobs.
However, in the dynamic cluster creation / deletion environment, it's challenging, I believe.

In the future DRA support, I think we need to consider how to support it for MultiKueue the same as TopologyAwareScheduling.

cc @mimowo

Reply:
Actually, I think MultiKueue may work ootb if users repeat the configuration in management and worker cluster. 
I'm ok with a note as non-goal, but maybe just a graduation point for beta is enough.

Reply:
Let's add MultiKueue to the Beta criteria.

Reply:
added.